examples/TestAsync.purs

```diff
@@ -0,0 +1,48 @@
+module Main where
+
+import Prelude
+import Data.Either
+import Data.Maybe
+import Control.Monad.Eff
+import Control.Monad.Eff.Console(log,CONSOLE())
+import Control.Monad.Eff.Exception
+
+import Node.FS
+import qualified Node.FS.Async as A
+import qualified Node.Path as FP
+import qualified Node.Buffer as B 
+
+-- exercise the file descriptor based async IO functions
+
+main :: forall eff . Eff (fs::FS,err::EXCEPTION,console::CONSOLE|eff) Unit
+main = do
+  let path1 = FP.concat( ["examples", "TestAsync.purs"] )
+      path2 = FP.concat( ["examples", "TestAsync.purs.partial"] )
+      buf = B.create 1000
+  A.fdOpen path1 R Nothing $ \v -> case v of
+    (Left err) -> log ("err:" ++ show err)
```

Comment:
> This was just meant to be a trivial demo to meet your request

Ah yes of course, sorry, disregard that then. In that case let's leave this as-is for now.

I certainly agree with you that a test suite would be nice. The only reason we don't, as far as I'm aware, is that nobody has gotten around to it yet. I might have a go soon.